### PR TITLE
4616: Ordering Party GET

### PR DESCRIPTION
**Pull request by RuthBerry96**
Story: [AB#4616](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/4616)
Task: [AB#7170](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7170)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs [hunk 1]
```diff
@@ -1,42 +1,71 @@
...
+using NHSD.BuyingCatalogue.Ordering.Application.Persistence;
+using NHSD.BuyingCatalogue.Ordering.Common.Constants;
 
 namespace NHSD.BuyingCatalogue.Ordering.Api.Controllers
 {
     [Route("api/v1/orders/{orderId}/sections/ordering-party")]
     [ApiController]
     [Produces("application/json")]
+    [Authorize(Policy = PolicyName.CanAccessOrders)]
     public sealed class OrderingPartyController : Controller
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs:16]: I dont mind changing it, but what actually is the difference?

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs [hunk 2]
```diff
@@ -1,42 +1,71 @@
...
+            if (order is null)
+            {
+                return NotFound();
+            }
+
+            var primaryOrganisationId = User.GetPrimaryOrganisationId();
+            if (primaryOrganisationId != order.OrganisationId)
+            {
+                return Forbid();
+            }
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs:39]: Okay, shall we make sure this comes up in the discussion on Friday?

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs [hunk 3]
```diff
@@ -1,42 +1,71 @@
...
+        private readonly IOrderRepository _orderRepository;
+
+        public OrderingPartyController(IOrderRepository orderRepository)
+        {
+            _orderRepository = orderRepository ?? throw new ArgumentNullException(nameof(orderRepository));
+        }
+
         [HttpGet]
-        public ActionResult Get(string orderId)
+        public async Task<ActionResult> GetAsync(string orderId)
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs:26]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Ordering Party Get 4616.feature [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+Scenario: 5. A buyer user cannot access the ordering-party for an organisation they don't belong to
+    Given the user is logged in with the Buyer role for organisation e6ea864e-ef1b-41aa-a4d5-04fc6fce0933
+    When the user makes a request to retrieve the ordering-party section with the ID C000014-01
+    Then a response with status code 403 is returned
+
+@4616
+Scenario: 6. Service Failure
+    Given the call to the database will fail
+    When the user makes a request to retrieve the ordering-party section with the ID C000014-01
+    Then a response with status code 500 is returned
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Ordering Party Get 4616.feature:59]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/AddressSteps.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+                var addressId = await address.InsertAsync<int>(_settings.ConnectionString);
+                addressDictionary.Add(address.Postcode, addressId);
+            }
+
+            _context[ScenarioContextKeys.AddressMapDictionary] = addressDictionary;
+        }
+
+        [Then(@"the Address section (.*) is returned")]
+        public async Task ThenTheAddressSectionOrganiationIsReturned(string section, Table table)
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/AddressSteps.cs:56]: *Actually didnt want organisation in the title, so removed it

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Utils/IntegrationTestDatabase.cs [hunk 6]
```diff
@@ -15,6 +15,8 @@ public static async Task ResetAsync(IConfiguration config)
             await databaseConnection.ExecuteAsync("ALTER ROLE db_datareader ADD MEMBER NHSD;");
             await databaseConnection.ExecuteAsync("ALTER ROLE db_datawriter ADD MEMBER NHSD;");
             await databaseConnection.ExecuteAsync("DELETE FROM [dbo].[Order];");
+            await databaseConnection.ExecuteAsync("DELETE FROM [dbo].[Address];");
+            await databaseConnection.ExecuteAsync("DELETE FROM [dbo].[Contact];");
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Utils/IntegrationTestDatabase.cs:19]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Builders/OrderBuilder.cs [hunk 7]
```diff
@@ -13,6 +13,10 @@ private OrderBuilder()
             {
                 OrderId = "C000014-01",
                 OrganisationId = Guid.NewGuid(),
+                OrganisationName = "Organisation Name",
+                OrganisationOdsCode = "Ods Code",
+                OrganisationAddress = new Address() { AddressId = 1, Line1 = "Line1", Line2 = "Line2", Line3 = "Line3", Town = "Leeds",County = "W Yorks", Postcode = "LS1 123", Country = "W YORKS"},
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Builders/OrderBuilder.cs:18]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSteps.cs [hunk 8]
```diff
@@ -111,13 +128,48 @@ private static object CreateOrders(JToken token)
...
+
+            public string OrganisationAddressPostcode { get; set; }
+
+            public string OrganisationContactEmail { get; set; }
+
+            public int? OrganisationBillingAddressId { get; set; }
+
+            public int? OrganisationContactId { get; set; }
 
             public int OrderStatusId { get; set; }
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSteps.cs:175]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+            var response = await controller.GetAsync("INVALID");
+            response.Should().BeEquivalentTo(new NotFoundResult());
+        }
+
+        [Test]
+        public async Task GetAsync_WrongOrganisationId_ReturnsForbidden()
+        {
+            var context = OrderingPartyTestContext.Setup();
+
+            var orderId = "C0000014-01";
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs:45]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bjss-chris) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs:16]: I've linked to the MS docs (click on [`ControllerBase`](https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-3.1#controllerbase-class)), which will go into more detail but essentially `Controller` adds support for views, which we don't need.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+            using var controller = context.OrderingPartyController;
+
+            var result = await controller.GetAsync(orderId);
+            result.Should().BeOfType<ForbidResult>();
+        }
+
+        [Test]
+        public async Task GetAsync_OrderIdExists_ReturnsTheOrderingParty()
+        {
+            var orderId = "C0000014-01";
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs:59]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+                    ControllerContext = new ControllerContext
+                    {
+                        HttpContext = new DefaultHttpContext { User = ClaimsPrincipal }
+                    }
+                };
+            }
+
+            internal Guid PrimaryOrganisationId { get; }
+
+            internal ClaimsPrincipal ClaimsPrincipal { get; }
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs:150]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/AddressEntity.cs [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+        public string Line3 { get; set; }
+        public string Line4 { get; set; }
+        public string Line5 { get; set; }
+        public string Town { get; set; }
+        public string County { get; set; }
+        public string Postcode { get; set; }
+        public string Country { get; set; }
+
+        protected override string InsertSql => $@"
+            INSERT INTO [dbo].[Address]
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/AddressEntity.cs:16]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/AddressEntity.cs [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+                @Line1,
+                @Line2,
+                @Line3,
+                @Line4,
+                @Line5,
+                @Town,
+                @County,
+                @Postcode,
+                @Country
+            )
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/AddressEntity.cs:39]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/AddressEntity.cs [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+            SELECT SCOPE_IDENTITY()";
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/AddressEntity.cs:40]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/AddressEntity.cs [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+        public string Line2 { get; set; }
+        public string Line3 { get; set; }
+        public string Line4 { get; set; }
+        public string Line5 { get; set; }
+        public string Town { get; set; }
+        public string County { get; set; }
+        public string Postcode { get; set; }
+        public string Country { get; set; }
+
+        protected override string InsertSql => $@"
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/AddressEntity.cs:15]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/ContactEntity.cs [hunk 16]
```diff
@@ -0,0 +1,27 @@
+﻿namespace NHSD.BuyingCatalouge.Ordering.Api.Testing.Data.Entities
+{
+    public sealed class ContactEntity : EntityBase
+    {
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public string Email { get; set; }
+        public string Phone { get; set; }
+
+        protected override string InsertSql => $@"
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/ContactEntity.cs:10]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/ContactEntity.cs [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public sealed class ContactEntity : EntityBase
+    {
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public string Email { get; set; }
+        public string Phone { get; set; }
+
+        protected override string InsertSql => $@"
+            INSERT INTO [dbo].[Contact]
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/ContactEntity.cs:11]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/ContactEntity.cs [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+                Email,
+                Phone
+            )
+            VALUES
+            (
+                @FirstName,
+                @LastName,
+                @Email,
+                @Phone
+            )
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/ContactEntity.cs:24]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/ContactEntity.cs [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+            SELECT SCOPE_IDENTITY()";
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/ContactEntity.cs:25]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs [hunk 20]
```diff
@@ -27,6 +37,11 @@ public sealed class OrderEntity : EntityBase
                 OrderId,
                 Description,
                 OrganisationId,
+                OrganisationName,
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:40]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs [hunk 21]
```diff
@@ -38,6 +53,11 @@ public sealed class OrderEntity : EntityBase
                 @OrderId,
                 @Description,
                 @OrganisationId,
+                @OrganisationName,
+                @OrganisationOdsCode,
+                @OrganisationAddressId,
+                @OrganisationBillingAddressId,
+                @OrganisationContactId,
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:60]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/EntityBuilder/OrderEntityBuilder.cs [hunk 22]
```diff
@@ -13,6 +13,9 @@ private OrderEntityBuilder()
             {
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/EntityBuilder/OrderEntityBuilder.cs:13]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.Persistence/Repositories/OrderRepository.cs [hunk 23]
```diff
@@ -22,13 +22,26 @@ public OrderRepository(ApplicationDbContext context)
...
         public async Task<IEnumerable<Order>> ListOrdersByOrganisationIdAsync(Guid organisationId)
         {
-            return await _context.Order.Include(x => x.OrderStatus).Where(o => o.OrganisationId == organisationId)
+            return await _context.Order
+                .Include(x => x.OrderStatus)
+                .Include(x => x.OrganisationAddress)
+                .Include(x => x.OrganisationContact)
+                .Where(o => o.OrganisationId == organisationId)
                 .ToListAsync();
         }
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Persistence/Repositories/OrderRepository.cs:31]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs [hunk 24]
```diff
@@ -0,0 +1,164 @@
...
+            result.Should().BeOfType<ForbidResult>();
+        }
+
+        [Test]
+        public async Task GetAsync_OrderIdExists_ReturnsTheOrderingParty()
+        {
+            var orderId = "C0000014-01";
+            var context = OrderingPartyTestContext.Setup();
+
+            var testData = CreateOrderingPartyTestData(orderId, context.PrimaryOrganisationId);
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs:62]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs [hunk 25]
```diff
@@ -1,42 +1,71 @@
...
+using NHSD.BuyingCatalogue.Ordering.Api.Extensions;
 using NHSD.BuyingCatalogue.Ordering.Api.Models;
+using NHSD.BuyingCatalogue.Ordering.Application.Persistence;
+using NHSD.BuyingCatalogue.Ordering.Common.Constants;
 
 namespace NHSD.BuyingCatalogue.Ordering.Api.Controllers
 {
     [Route("api/v1/orders/{orderId}/sections/ordering-party")]
     [ApiController]
     [Produces("application/json")]
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs:14]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs [hunk 26]
```diff
@@ -0,0 +1,164 @@
...
+            response.Should().BeEquivalentTo(new NotFoundResult());
+        }
+
+        [Test]
+        public async Task GetAsync_WrongOrganisationId_ReturnsForbidden()
+        {
+            var context = OrderingPartyTestContext.Setup();
+
+            var orderId = "C0000014-01";
+            var testData = CreateOrderingPartyTestData(orderId, Guid.NewGuid());
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs:46]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSteps.cs [hunk 27]
```diff
@@ -1,187 +1,187 @@
...
+        {
+            var orders = (await _response.ReadBodyAsJsonAsync()).Select(CreateOrders);
+            orders.Count().Should().Be(0);
+        }
+
+        [Then(@"the order is created in the database with orderId (.*) and data")]
+        public async Task ThenTheOrderIsCreatedInTheDatabase(string orderId, Table table)
+        {
+            var actual = await OrderEntity.FetchOrderByOrderId(_settings.ConnectionString, orderId);
+            table.CompareToInstance<OrderEntity>(actual);
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSteps.cs:101]: No essential, but we can omit the type here (`OrderEntity`).

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs [hunk 28]
```diff
@@ -147,7 +147,7 @@ private OrderingPartyTestContext()
 
             internal Guid PrimaryOrganisationId { get; }
 
-            internal ClaimsPrincipal ClaimsPrincipal { get; }
+            private ClaimsPrincipal ClaimsPrincipal { get; }
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs:150]: Not essential, but now that this is `private` ideally it should be below the `internal` properties (based on ordering from most accessible > least accessible).

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs [hunk 29]
```diff
@@ -63,7 +63,7 @@ public sealed class OrderEntity : EntityBase
                 @LastUpdated,
                 @LastUpdatedBy,
                 @LastUpdatedByName
-            )";
+            );";
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:66]: It looks like this file has mixed line endings too – can you fix them?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs (hunk 28 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs:150]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs (hunk 29 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:66]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSteps.cs (hunk 27 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSteps.cs:101]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs [hunk 30]
```diff
@@ -1,99 +1,99 @@
...
+                @LastUpdatedByName
+            );";
+
         public static async Task<string> FetchOrderDescriptionFromOrderId(string connectionString, string orderId)
-        {
-            return (await SqlRunner.QueryFirstAsync<string>(connectionString, $@"SELECT
-                         [Description]
-                         FROM [Order]
+        {
+            return (await SqlRunner.QueryFirstAsync<string>(connectionString, $@"SELECT
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:70]: As we're not actually doing any string interpolation here we can omit the `$`.

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs [hunk 31]
```diff
@@ -1,99 +1,99 @@
...
+                         [Description]
+                         FROM [Order]
+                         WHERE [OrderId] = @orderId", new { orderId }));
+        }
+
+        public static async Task<string> FetchLastUpdatedByNameFromOrderId(string connectionString, string orderId)
+        {
+            return (await SqlRunner.QueryFirstAsync<string>(connectionString, $@"SELECT
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:78]: As we're not actually doing any string interpolation here we can omit the `$`.

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs [hunk 32]
```diff
@@ -1,99 +1,99 @@
...
-                          [Created],
-                          [LastUpdated],
-                          [LastUpdatedBy],
-                          [LastUpdatedByName]
-                         FROM [Order]
-                         WHERE [OrderId] = @orderId", new { orderId }));
-        }
-    }
-}
+            return (await SqlRunner.QueryFirstAsync<OrderEntity>(connectionString, $@"SELECT
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:86]: As we're not actually doing any string interpolation here we can omit the `$`.

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs (hunk 30 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs (hunk 31 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs (hunk 32 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs [hunk 33]
```diff
@@ -67,23 +67,23 @@ public sealed class OrderEntity : EntityBase
...
+            return (await SqlRunner.QueryFirstAsync<string>(connectionString, @"SELECT
                          [LastUpdatedByName]
                          FROM [Order]
                          WHERE [OrderId] = @orderId", new { orderId }));
         }
 
         public static async Task<OrderEntity> FetchOrderByOrderId(string connectionString, string orderId)
         {
-            return (await SqlRunner.QueryFirstAsync<OrderEntity>(connectionString, $@"SELECT
+            return (await SqlRunner.QueryFirstAsync<OrderEntity>(connectionString, @"SELECT
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:86]: One last request (I promise) – can you do a little SQL tidying? Remove square brackets where not required, add in schema identifiers (`dbo`) where not present and add terminating semicolons?

Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/Entities/OrderEntity.cs:86]: 👍

**Pull request merged**